pkg/controller/hostpathprovisioner/prometheus.go

```diff
@@ -181,167 +185,6 @@ func (r *ReconcileHostPathProvisioner) deletePrometheusResources(namespace strin
-		recordRules = append(recordRules, generateRecordRule(rrd.Name, rrd.Expr))
-	}
-
-	return recordRules
-}
-
-func getAlertRules(runbookURLTemplate string) []promv1.Rule {
-	return []promv1.Rule{
-		generateAlertRule(
-			"HPPOperatorDown",
-			"kubevirt_hpp_operator_up == 0",
-			"5m",
-			map[string]string{
-				"summary":     "Hostpath Provisioner operator is down",
-				"runbook_url": fmt.Sprintf(runbookURLTemplate, "HPPOperatorDown"),
-			},
-			map[string]string{
-				severityAlertLabelKey:     "warning",
-				healthImpactAlertLabelKey: "critical",
-				partOfAlertLabelKey:       partOfAlertLabelValue,
-				componentAlertLabelKey:    componentAlertLabelValue,
-			},
-		),
-		generateAlertRule(
-			"HPPNotReady",
-			"kubevirt_hpp_cr_ready == 0",
-			"5m",
-			map[string]string{
-				"summary":     "Hostpath Provisioner is not available to use",
-				"runbook_url": fmt.Sprintf(runbookURLTemplate, "HPPNotReady"),
-			},
-			map[string]string{
-				severityAlertLabelKey:     "warning",
-				healthImpactAlertLabelKey: "critical",
-				partOfAlertLabelKey:       partOfAlertLabelValue,
-				componentAlertLabelKey:    componentAlertLabelValue,
-			},
-		),
-		generateAlertRule(
-			"HPPSharingPoolPathWithOS",
-			"kubevirt_hpp_pool_path_shared_with_os == 1",
-			"1m",
-			map[string]string{
-				"summary":     "HPP pool path sharing a filesystem with OS, fix to prevent HPP PVs from causing disk pressure and affecting node operation",
-				"runbook_url": fmt.Sprintf(runbookURLTemplate, "HPPSharingPoolPathWithOS"),
-			},
-			map[string]string{
-				severityAlertLabelKey:     "warning",
-				healthImpactAlertLabelKey: "warning",
-				partOfAlertLabelKey:       partOfAlertLabelValue,
-				componentAlertLabelKey:    componentAlertLabelValue,
-			},
-		),
-	}
-}
-
-func createPrometheusRule(namespace string) *promv1.PrometheusRule {
-	labels := getRecommendedLabels()
-	labels[PrometheusLabelKey] = PrometheusLabelValue
-
-	runbookURLTemplate := getRunbookURLTemplate()
-
-	return &promv1.PrometheusRule{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: promv1.SchemeGroupVersion.String(),
-			Kind:       "PrometheusRule",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      ruleName,
-			Namespace: namespace,
-			Labels:    labels,
-		},
-		Spec: promv1.PrometheusRuleSpec{
-			Groups: []promv1.RuleGroup{
-				{
-					Name:  "hpp.rules",
-					Rules: append(getRecordRules(namespace), getAlertRules(runbookURLTemplate)...),
-				},
-			},
-		},
-	}
-}
-
-func createPrometheusRole(namespace string) *rbacv1.Role {
-	labels := getRecommendedLabels()
-	labels[PrometheusLabelKey] = PrometheusLabelValue
-
-	return &rbacv1.Role{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      rbacName,
-			Namespace: namespace,
-			Labels:    labels,
-		},
-		Rules: []rbacv1.PolicyRule{
-			{
-				APIGroups: []string{
-					"",
-				},
-				Resources: []string{
-					"services",
-					"endpoints",
-					"pods",
-				},
-				Verbs: []string{
-					"get", "list", "watch",
```

Comment:
@machadovilaca not sure if we need to fix operator-observability repo, since it's not including watch verb, see: https://github.com/machadovilaca/operator-observability/blob/1643dc40aefa852d9cd12610ed3ddfc0e3e686d6/pkg/operatorrules/rbac.go#L19

Reply:
Let's first make sure it is needed. I think kubevirt/kubevirt only uses  `"get", "list"`, but might be a different approach